app/assets/javascripts/repositories/repository_datatable.js

```diff
@@ -32,27 +50,429 @@ var RepositoryDatatable = (function(global) {
+
+          if (val) {
+            appendInput(form, val, 'header_ids[]');
+          }
+        });
+
+        // Append records
+        $.each(rowsSelected, function(index, rowId) {
+          appendInput(form, rowId, 'row_ids[]');
+        });
+      }
+    });
+  }
+
+  function disableCheckboxToggleOnAssetDownload() {
+    $('.file-preview-link').on('click', function(ev) {
+      ev.stopPropagation();
+    });
+  }
+
+  // Return td element with content
+  function createTdElement(content) {
+    var td = document.createElement('td');
+    td.innerHTML = content;
+    return td;
+  }
+
+  function initialListItemsRequest(columnId) {
+    var massageResponse = [];
+    $.ajax({
+      url: $(TABLE_ID).data('list-items-path'),
+      type: 'POST',
+      dataType: 'json',
+      async: false,
+      data: {
+        q: '',
+        column_id: columnId
+      }
+    }).done(function(data) {
+      $.each(data.list_items, function(index, el) {
+        massageResponse.push([el.id, el.data]);
+      });
+    });
+    return listItemDropdown(massageResponse, '-1', columnId);
+  }
+
+  function initSelectPicker() {
+    $('.selectpicker')
+      .selectpicker({ liveSearch: true })
+      .ajaxSelectPicker({
+        ajax: {
+          url: $(TABLE_ID).data('list-items-path'),
+          type: 'POST',
+          dataType: 'json',
+          data: function() {
+            var params = {
+              q: '{{{q}}}',
+              column_id: $(this.valueOf().plugin.$element).attr('column_id')
+            };
+
+            return params;
+          }
+        },
+        locale: {
+          emptyTitle: 'Nothing selected'
+        },
+        preprocessData: function(data) {
+          var items = [];
+          if (Object.prototype.hasOwnProperty.call(data, 'list_items')) {
+            items.push({
+              value: '-1',
+              text: '',
+              disabled: false
+            });
+            $.each(data.list_items, function(index, el) {
+              items.push(
+                {
+                  value: el.id,
+                  text: el.data,
+                  disabled: false
+                }
+              );
+            });
+          }
+          return items;
+        },
+        emptyRequest: true,
+        clearOnEmpty: false,
+        preserveSelected: false
+      }).on('change.bs.select', function(el) {
+        $(this).closest('td').attr('list_item_id', el.target.value);
+        $(this).closest('td').attr('column_id', $(this).attr('column_id'));
+      })
+      .trigger('change.bs.select');
+  }
+
+  // Adjust columns width in table header
+  function adjustTableHeader() {
+    TABLE.columns.adjust();
+    $('.dropdown-menu').parent()
+      .on('shown.bs.dropdown hidden.bs.dropdown', function() {
+        TABLE.columns.adjust();
+      });
+  }
+
+  // Clear all has-error tags
+  function clearAllErrors() {
+    // Remove any validation errors
+    $(selectedRecord).find('.has-error').each(function() {
```

Comment:
It also can be done without cycle:
`$(selectedRecord).find('.has-error').removeClass('has-error').find('span').remove()`